go/parquet/internal/encoding/boolean_encoder.go

```diff
@@ -87,3 +89,53 @@ func (enc *PlainBooleanEncoder) FlushValues() (Buffer, error) {
 
 	return enc.sink.Finish(), nil
 }
+
+type RleBooleanEncoder struct {
+	encoder
+
+	bufferedValues []bool
+}
+
+func (RleBooleanEncoder) Type() parquet.Type {
+	return parquet.Types.Boolean
+}
+
+func (enc *RleBooleanEncoder) Put(in []bool) {
+	enc.bufferedValues = append(enc.bufferedValues, in...)
+}
+
+func (enc *RleBooleanEncoder) PutSpaced(in []bool, validBits []byte, validBitsOffset int64) {
+	bufferOut := make([]bool, len(in))
+	nvalid := spacedCompress(in, bufferOut, validBits, validBitsOffset)
+	enc.Put(bufferOut[:nvalid])
+}
+
+func (enc *RleBooleanEncoder) EstimatedDataEncodedSize() int64 {
+	const rleLengthInBytes = 4
+	return rleLengthInBytes + int64(enc.maxRleBufferSize())
+}
+
+func (enc *RleBooleanEncoder) maxRleBufferSize() int {
+	return utils.MaxBufferSize(1, len(enc.bufferedValues))
```

Comment:
Not really, the C++ implementation calls `RlePreserveBufferSize` which is much more sophisticated than this (and, if I'm not mistaken, it would return `len(enc.bufferedValues) / 4` for booleans).

Reply:
`RlePreserveBufferSize` is implemented as `MaxBufferSize(bit_width, num_values) + MinBufferSize(bit_width)` I just skipped the middle `RlePreserveBufferSize` here, though I should add the `+ MinBufferSize(1)` here so I will.

Reply:
Are you saying that `utils.MaxBufferSize` is a RLE-specific utility? The name is not really informative...

Reply:
renamed the functions